docs/upgrade-to-7.md

```diff
@@ -0,0 +1,132 @@
+# Upgrade to v7
+
+This guide will take you through updating `@gasket/*` packages to `7.x`.
+
+## Update Dependency Versions
+
+Update all `@gasket/` scoped packages to the v7 major version.
+
+This is not an exhaustive list, but rather a sampling of dependencies to
+demonstrate what to look for:
+
+```diff
+"dependencies": {
+-    "@gasket/fetch": "^5.0.2",
++    "@gasket/fetch": "^7.0.0",
+-    "@gasket/data": "^5.5.0",
++    "@gasket/data": "^7.0.0",
+-    "@gasket/plugin-workbox": "^5.4.1",
++    "@gasket/plugin-workbox": "^7.0.0",
+}
+```
+
+## Code Removal
+
+### @gasket/gasket-cli
+
+Removed support for deprecated `--npmconfig` flag. [(#647)]
+
+### @gasket/plugin-workbox
+
+Remove deprecated `assetPrefix` config support. Use `basePath` instead [(#661)]
+
+### @gasket/plugin-nextjs
+
+Remove deprecated `next` config support. Use `nextConfig` instead[(#655)]
+
+### @gasket/gasket-utils
+
+Remove deprecated `applyEnvironmentOverrides`. [(#649)]
+
+### @gasket/plugin-intl
+
+Remove deprecated support for `languageMap`, `defaultLanguage`, `assetPrefix` config options. [(#666)]
+
+### @gasket/plugin-webpack
+
+- Remove deprecated lifecycles.
+- Remove `webpackMerge` util.
+
+[(#665)]
+
+### @gasket/plugin-elastic-apm
+
+- Remove deprecated config support.
+- Do not start in preboot, log warning if not started.
+
+[(#668)]
+
+## @gasket/gasket-plugin-cypress
+
+Update Cypress version to 12.3.0. [(#660)]
+
+## Set Docusaraus as The Default Docs Generator
+
+Replace `docsify` with `docusaurus` for gasket docs. [(#673)]
+
+## Align Lifecycles to use Context Object in Params
+
+We had some lifecycles that did not conform to the context object pattern we have in place for many other lifecycles.
+
+The reason for utilizing this context object is to enable the execution of these lifecycle methods under two distinct scenarios:
+
+1. During build time, when there is no request object available.
+2. During run time, when the request object becomes available.
+
+Affected lifecycles:
+
+- `appRequestConfig`
```

Comment:
Another big change was renaming config plugin to response-data plugin ([#680](https://github.com/godaddy/gasket/pull/680)). Although yes, this lifecycle context has been normalized, it is under a new name now: `responseData` ([doc change](https://github.com/godaddy/gasket/pull/680/files#diff-cffc29cc7d0e75f6e9ea8cde81073208211edbea7c8de39c341574d318541304R243)).

You could probably leave it out of this section, but add a new section for the config -> response-data changes.